starboard/nplb/dmp_file_selector.cc

```diff
@@ -0,0 +1,126 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "starboard/nplb/dmp_file_selector.h"
+
+#include "starboard/shared/starboard/player/video_dmp_reader.h"
+
+namespace starboard {
+namespace nplb {
+
+using shared::starboard::player::video_dmp::VideoDmpReader;
+
+DmpFileSelector::DmpFileSelector() {
+  if (SbMediaCanPlayMimeAndKeySystem(
+          "video/mp4; codecs=\"av01.0.16M.08\"; width=7680; height=4320", "") ==
+      kSbMediaSupportTypeProbably) {
+    supports_av1_ = true;
+    type_ = kDeviceType8k;
```

Comment:
`type_` is assigned but not used, consider removing.

Reply:
Why didn't our linters or `clang-tidy` catch that ?